src/hooks/use-tabbing-detect.ts

```diff
@@ -0,0 +1,40 @@
+import { useEffect } from 'react'
+
+// https://medium.com/hackernoon/removing-that-ugly-focus-ring-and-keeping-it-too-6c8727fefcd2
+
+const tabbingClass = 'user-is-tabbing'
+
+function handleFirstTab(event: KeyboardEvent) {
+  if (event.code === `Tab`) {
+    reset()
+    document.body.classList.add(tabbingClass)
+    window.removeEventListener('keydown', handleFirstTab)
+  }
+}
+
+function handleClick() {
+  reset()
+  window.addEventListener('keydown', handleFirstTab)
+}
+
+function reset() {
+  document.body.classList.remove(tabbingClass)
+  window.removeEventListener('keydown', handleClick)
+  window.removeEventListener('mousedown', handleFirstTab as EventListener)
```

Comment:
these look inverted. like, `keydown` with the click handler, and vice versa. is it intentional?

Reply:
nup, fixed it